### PR TITLE
Ancestry Script

### PR DESCRIPTION
**Pull request by smlmbrt**
Making this a separate branch so that I don't keep pushing new images (that include new packages like scikit-learn). This is to replace the example notebook which will only be rendered later!

### CODE REVIEW

File: pgscatalog_utils/ancestry/ancestry_analysis.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import textwrap
+import logging
+import glob
+import os
+
+import pandas as pd
+
+import pgscatalog_utils.config as config
+from pgscatalog_utils.ancestry.read import read_pcs, read_pgs, extract_ref_psam_cols
+from pgscatalog_utils.ancestry.tools import assign_ancestry, _assign_method_threshold, choose_pval_threshold, \
```
Comment (nebfield) [inline pgscatalog_utils/ancestry/ancestry_analysis.py:11]: function names that start with `_` are normally for internal use (i.e. inside `pgscatalog_utils.ancestry.tools`)

File: pgscatalog_utils/ancestry/ancestry_analysis.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    args = _parse_args()
+    config.set_logging_level(args.verbose)
+    config.OUTDIR = args.outdir
+
+    # Load PCA data
+    maxPCs = max([10, args.nPCs_assignment, args.nPCs_normalization])  # save memory by not using all PCs
+    loc_ref_pcs = glob.glob('*{}*.pcs'.format(args.d_ref))
+    reference_df = read_pcs(loc_pcs=loc_ref_pcs, dataset=args.d_ref,
+                            loc_related_ids=args.ref_related, nPCs=maxPCs)
+    loc_ref_psam = glob.glob('GRCh38_{}_ALL.psam'.format(args.d_ref))[0]
```
Comment (nebfield) [inline pgscatalog_utils/ancestry/ancestry_analysis.py:27]: will this break if reference data are `GRCh37`?

Comment (smlmbrt) [reply pgscatalog_utils/ancestry/ancestry_analysis.py:27]: Oops, that wasn't meant to be hardcoded. Can switch to `'*{}*.psam'.format(args.d_ref)`

File: pgscatalog_utils/ancestry/ancestry_analysis.py (hunk 1 above)
Comment (smlmbrt) [reply pgscatalog_utils/ancestry/ancestry_analysis.py:11]: That makes sense, happy to change it

File: pgscatalog_utils/ancestry/ancestry_analysis.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    reference_df = extract_ref_psam_cols(loc_ref_psam, args.d_ref, reference_df, keepcols=[args.ref_label])
+
+    loc_target_sscores = glob.glob('*{}*.pcs'.format(args.d_target))
```
Comment (nebfield) [inline pgscatalog_utils/ancestry/ancestry_analysis.py:30]: would it better for some arguments to take a list of paths like other utilities?

Comment (smlmbrt) [reply pgscatalog_utils/ancestry/ancestry_analysis.py:30]: Yeah, that probably makes sense and easy to coordinate with the nextflow module? If they all have the same naming format we can find the dataset name in it and remove that argument.

File: pyproject.toml [hunk 4]
```diff
@@ -35,6 +38,9 @@ pysqlar = "^0.1.2"
 memory-profiler = "^0.60.0"
 matplotlib = "^3.6.0"
 
+[tool.poetry.group.dev.dependencies]
+jupyter = "^1.0.0"
```
Comment (nebfield) [inline pyproject.toml:42]: do we still need Jupyter here?

**Review by nebfield — COMMENTED** [review]
just a few tiny questions about arguments, imports, and dependencies

File: pyproject.toml (hunk 4 above)
Comment (nebfield) [reply pyproject.toml:35]: I moved the dev dependencies to the dev group, because the old method was deprecated

**Pull request merged**